beacon-chain/slasher/service.go

```diff
@@ -3,27 +3,56 @@ package slasher
 
 import (
 	"context"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/shared/event"
 )
 
-// Service --
+// ServiceConfig for the slasher service in the beacon node.
+// This struct allows us to specify required dependencies and
+// parameters for slasher to function as needed.
+type ServiceConfig struct {
+	*Parameters
+	IndexedAttsFeed *event.Feed
+	Database        db.Database
+}
+
+// Service defining a slasher implementation as part of
+// the beacon node, able to detect eth2 slashable offenses.
 type Service struct {
+	params          *Parameters
+	serviceCfg      *ServiceConfig
+	indexedAttsChan chan *ethpb.IndexedAttestation
+	ctx             context.Context
+	cancel          context.CancelFunc
 }
 
-// NewService --
-func NewService(ctx context.Context) (*Service, error) {
-	return &Service{}, nil
+// NewService instantiates a new slasher from configuration values.
+func NewService(ctx context.Context, srvCfg *ServiceConfig) (*Service, error) {
```

Comment:
```suggestion
func New(ctx context.Context, srvCfg *ServiceConfig) (*Service, error) {
```

or else @rkapka will complain : )